packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -277,7 +303,8 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
     isFocused,
     setFocused,
     selectedItem,
-    collection: filteredCollection,
+    // TODO: should there be a prop that controls this behavior for react-stately consumers only?
+    collection: showAllItemsTracker && (props.items === undefined) ? originalCollection : filteredCollection,
```

Comment:
open question, perhaps some people wouldn't want the display all items behavior on menu open for uncontrolled comboboxes

Reply:
I think for now, no prop. We can possibly add something if it comes up.